Sources/Sentry/Profiling/SentryLegacyProfiler.mm

```diff
@@ -0,0 +1,92 @@
+#import "SentryLegacyProfiler.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import "SentryLog.h"
+#    import "SentryMetricProfiler.h"
+#    import "SentryProfiledTracerConcurrency.h"
+#    import "SentryProfiler+Private.h"
+#    include <mutex>
+
+#    pragma mark - Private
+
+namespace {
+/** @warning: Must be used from a synchronized context. */
+std::mutex _unsafe_gLegacyProfilerLock;
+
+/** @warning: Must be used from a synchronized context. */
+SentryProfiler *_Nullable _unsafe_gLegacyProfiler;
+
+/** @warning: Must be called from a synchronized context. */
+BOOL
+_unsafe_isRunning(void)
+{
+    return _unsafe_gLegacyProfiler != nil && [_unsafe_gLegacyProfiler isRunning];
+}
+} // namespace
+
+@implementation SentryLegacyProfiler
+
+#    pragma mark - Public
+
++ (BOOL)startWithTracer:(SentryId *)traceId
+{
+    std::lock_guard<std::mutex> l(_unsafe_gLegacyProfilerLock);
+
+    if (_unsafe_isRunning()) {
+        SENTRY_LOG_DEBUG(@"A legacy profiler is already running.");
+        trackProfilerForTracer(_unsafe_gLegacyProfiler, traceId);
+        // record a new metric sample for every concurrent span start
+        [_unsafe_gLegacyProfiler.metricProfiler recordMetrics];
+        return YES;
+    }
+
+    _unsafe_gLegacyProfiler = [[SentryProfiler alloc] initWithMode:SentryProfilerModeLegacy];
```

Comment:
This is the only line that actually changed when moving this code here from SentryProfiler.